src/main/java/org/edumips64/core/Converter.java

```diff
@@ -617,5 +567,70 @@ public static long powLong(int base, long exp) {
 
     return ret;
   }
+
+  /**
+   * Determines if a string contains only 0's and 1's
+   * @param bits any string
+   * @return false if bits contains character that is not '0' and not '1', true otherwise
+   */
+  private static boolean isBinaryString(String bits) throws IrregularStringOfBitsException{
+    char bit;
+    for (int i = 0; i < bits.length(); i++) {
+      bit = bits.charAt(i);
+      if (bit != '0' && bit != '1') {
+        throw new IrregularStringOfBitsException();
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Determines if msb is '1' and all the rest '0'
+   * @param bits any string of only 0's and 1's
+   * @return true if msb is '1' and all the rest '0', false otherwise
+   */
+  private static boolean isOverflow(String bits){
+    if(bits.charAt(0) != '1'){
+      return false;
+    }
+    for (int i = 1; i < bits.length(); i++) {
+      if (bits.charAt(i) != '0') {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Determines the integer value of an unsigned string of bits
+   * bits any string of only 0's and 1's
+   * @return integer value of the bit string
+   */
+  private static int getUnsignedIntValue(String bits){
+    int value = 0;
+    int i = 0;
+    for (int j = bits.length() - 1; j >= 0; j--, i++) {
+      if (bits.charAt(j) == '1') {
+        value += (int) Math.pow(2.0, (double) i);
```

Comment:
It's better to use shift operators here. Instead of

```java
value += (int) Math.pow(2.0, (double) i);
```

it's better to have
```java
value += 1 << i;
```

The left-shift operator is equivalent to 2 to the n-th power (`2 ^ i == 1 << i`), and has the advantage of not requiring conversions to floating-point and back, and also avoids any expensive operations that happen inside `Math.pow`. The left-shift operator is a single assembly instruction :)